GenumKit/Parsers/FontsFileParser.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  FontsFileParser.swift
+//  Pods
+//
+//  Created by Derek Ostrander on 3/7/16.
+//
+//
+
+import Foundation
+import AppKit.NSFont
+
+typealias Font = (name: String, variant: String)
+
+extension CFString {
+    var alphanumericString: String {
+         return (self as String).componentsSeparatedByCharactersInSet(NSCharacterSet.alphanumericCharacterSet().invertedSet).joinWithSeparator("")
+    }
+}
+
+extension CTFont {
+    static func loadFont(path: String) -> Font? {
+        let inData = NSData(contentsOfFile: path)
+        var error: Unmanaged<CFError>?
+        guard
+            let provider: CGDataProvider = CGDataProviderCreateWithCFData(inData),
```

Comment:
It's more efficient to use `CGDataProviderCreateWithURL` than read the whole `NSData(contentsOfFile:)` and use `CGDataPrividerCreateWithCFData`. Because `CGDataProviderCreateWithURL` will only reference the URL then only read the amount of data needed by further operations, avoiding reading the while file data if not needed.

Besides, I'm wondering if all that code is necessary. It seems quite CPU intensive to read the whole font data, create a CGFont from it, register it with the Font Manager (whereas we don't intent to use the font in a graphic context, only to extract its name and variations), then create a `CTFont` from there only to extract the metadata. I'm sure there are less intensive ways to do that?